lib/smart_proxy_host_reports/puppet_processor.rb

```diff
@@ -122,4 +122,17 @@ def spool_report
     end
     SpooledHttpClient.instance.spool(report_id, payload)
   end
+
+  private
+
+  def process_statuses
+    stats = @body["metrics"]["resources"]["values"].collect { |s| [s[0], s[2]] }
+      .to_h
+    {
+      applied: stats["changed"] + stats["corrective_change"],
+      failed: stats["failed"] + stats["failed_to_restart"],
+      pending: stats["scheduled"],
+      other: stats["restarted"] + stats["skipped"] + stats["out_of_sync"],
```

Comment:
The problem why tests fail is that you use Ruby symbols here, for the best performance I am using strings everywhere since JSON deserialization is into strings by default.